.github/workflows/build.yml

```diff
@@ -0,0 +1,59 @@
+name: Build
+
+on:
+    push:
+        branches: [main, next, master, 2.5, 3.4, 4.3, 5.1, 6.2]
+    pull_request:
+    release:
+        types: [created]
+    schedule:
+        -   cron: '0 4 * * *'
+
+jobs:
+    tests:
+        runs-on: ubuntu-latest
+        name: Build and test
+        strategy:
+            matrix:
+                php: [7.3, 7.4]
+                composer-flags:
+                include:
+                    -   php: 7.3
+                        composer-flags: --prefer-lowest
+                    -   php: 7.3
+                        composer-flags: --stability dev
+
+        env:
+            COMPOSER_ROOT_VERSION: dev-master
+
+        steps:
+            -   uses: actions/checkout@v2
+
+            -   name: Setup PHP
+                uses: shivammathur/setup-php@v2
+                with:
+                    php-version: "${{ matrix.php }}"
+                    ini-values: "phar.readonly=0"
+
+            -   name: Install dependencies
+                run: composer update ${{ matrix.composer-flags }}
+
+            -   name: Run tests (phpspec)
+                run: bin/phpspec run --format=dot
+
+            -   name: Run tests (phpunit)
+                run: ./vendor/bin/phpunit -v
+
+            -   name: Run tests (behat)
+                run: ./vendor/bin/behat --format=progress
+
+            -   name: Build PHAR
+                run: |
+                    composer config platform.php 7.3.0 &&
+                    ln -s `which composer` composer.phar &&
+                    make phpspec.phar
+
+            -   uses: actions/upload-artifact@v1
```

Comment:
this should run only for release runs

Reply:
also, probably not for all versions of the matrix. Instead, we should have a `build phar` job which depends on all tests jobs and deals with the phar when tests are successful.

Reply:
Yes, I planned to only upload it for 7.3 (or whatever the current minimum is).

I think it should run on every build, not just on release builds. It's useful to be able to download the phar for any build.

Attaching the file to the release is still left to do.